src/components/modal/Modal.stories.tsx

```diff
@@ -0,0 +1,81 @@
+import type {Meta, StoryObj} from '@storybook/react-vite';
+import {useState} from 'react';
+import type {ArgsStoryFn, Renderer} from 'storybook/internal/csf';
```

Comment:
which part of `internal` in this import is not clear?

Aren't you better of implementing this as a `import type {Decorator} from `@storybook/react-vite';`? 

You can then also grab the `useArgs` hook from the [story context](https://storybook.js.org/docs/writing-stories/decorators#context-for-mocking), instead of needing `useState` here, because changing the `isOpen` arg of the story will not have any effect when the modal open/closed state is tracked by `useState`.